Sources/PenguinParallel/ConcurrencyPlatform.swift

```diff
@@ -158,7 +160,7 @@ public struct TypedThreadLocalStorage<Underlying: RawThreadLocalStorage> {
 
   /// Token used to index into the thread local storage.
   public struct Key<Value: AnyObject> {
-    fileprivate let key: Underlying.Key
+    fileprivate var key: Underlying.Key
```

Comment:
Note: if `Underlying.Key` is a `class`, then I think this change can be reverted. WDYT?

Reply:
We can revert this change even without that - I call that out in the commit message with an explanation of _why_ I changed this.  It's purely for detection purposes.  If we say that the user is responsible for knowing when they have invalidated the TLS slot, this can remain a `let` binding.  One could argue that the destroy is an explicit action - most usage will be for the lifetime of the program, in which case, you just leak the slot which is irrelevant because the application is terminating and the resources will be released as a byproduct.  But, if you want to change the underlying `Key` to class even outside of that reasoning, I can certainly do that.